Testnet/NFTStore/NFTStore/NFTStore.cs

```diff
@@ -86,20 +64,40 @@ public void CancelSale(Address contract, ulong tokenId)
         Log(new TokenSaleCanceledLog { Contract = contract, TokenId = tokenId, Seller = saleInfo.Seller, Order = NextId++ });
     }
 
-    private bool IsApprovedForAll(Address contract, Address tokenOwner)
+    public bool OnNonFungibleTokenReceived(Address operatorAddress, Address fromAddress, ulong tokenId, byte[] data)
     {
-        var result = Call(contract, 0, "IsApprovedForAll", new object[] { tokenOwner, Message.Sender });
+        EnsureNotPayable();
 
-        Assert(result.Success, "IsApprovedForAll method call failed.");
+        var seller = fromAddress == Address.Zero ? operatorAddress : fromAddress;
```

Comment:
Does the `operatorAddress` need an `Addrss.Zero` check?

Reply:
I think it is not necessary check. There will be always an operator address actually. But fromAddress will be a zero when it is minted before this call.